test/NovaVaultV2.t.sol

```diff
@@ -297,4 +297,92 @@ contract NovaVaultV2Test is Test {
         assert(successWithdraw);
         assertEq(underlyingAmount, IERC20(underlyingAddress).balanceOf(alice));
     }
+
+    function testNovaVaultV2DoubleDeposit() public {
+        bool fromStableTosDai = true;
+        bool isStableFirst = true;
+
+        (uint160 sqrtPriceX96, , , , , ) = veloPool.slot0();
+        uint160 num = fromStableTosDai ? 95 : 105;
+        int256 sign = isStableFirst ? int256(1) : int256(-1);
+
+        LibSwap.SwapData[] memory swapData = new LibSwap.SwapData[](2);
+        swapData[0] = LibSwap.SwapData(
+            address(veloPool),
+            address(veloPool),
+            underlyingAddress,
+            sDAI,
+            aliceUnderlyingAmount,
+            abi.encodeWithSelector(
+                veloPool.swap.selector,
+                address(vault),
+                fromStableTosDai,
+                sign * int256(aliceUnderlyingAmount),
+                (num * sqrtPriceX96) / 100,
+                ""
+            ),
+            true
+        );
+
```

Comment:
you should define `swapData[1] = ...` just here, otherwise it will considered empty in the function `vault.deposit(...)` below where you pass `swapData`